### ISSUE TITLE
Endless loop when getting bucket lifecycle

### ISSUE THREAD

**Issue by v3rm0n** [localstack/localstack#3029]
# Type of request: This is a ...

[x ] bug report
[ ] feature request

# Detailed description
When creating a bucket, putting bucket website configuration and then getting the bucket lifecycle information, localstack goes into endless loop and never returns.

## Expected behavior

I get the bucket lifecycle

## Actual behavior

Command hangs and in localstack logs I see:
```
2020-09-25 12:57:22,201:API: 172.17.0.2 - - [25/Sep/2020 12:57:22] "PUT /test HTTP/1.1" 200 -                                                                                                                                                           
2020-09-25 12:57:37,011:API: 172.17.0.2 - - [25/Sep/2020 12:57:37] "HEAD /test HTTP/1.1" 200 -              
2020-09-25 12:57:37,021:API: 172.17.0.2 - - [25/Sep/2020 12:57:37] "HEAD /test HTTP/1.1" 200 -          
2020-09-25 12:57:37,030:API: 172.17.0.2 - - [25/Sep/2020 12:57:37] "GET /test?website HTTP/1.1" 404 -    
2020-09-25 12:59:28,895:API: 172.17.0.2 - - [25/Sep/2020 12:59:28] "PUT /test?website HTTP/1.1" 200 -       
2020-09-25 12:59:36,568:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "HEAD /test HTTP/1.1" 200 -          
2020-09-25 12:59:36,577:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "HEAD /test HTTP/1.1" 200 -           
2020-09-25 12:59:36,585:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "GET /test?website HTTP/1.1" 200 -       
2020-09-25 12:59:36,596:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "GET /test/index.html HTTP/1.1" 404 -2020-09-25 12:59:36,603:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "HEAD /test HTTP/1.1" 200 -           
2020-09-25 12:59:36,612:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "GET /test?website HTTP/1.1" 200 -
2020-09-25 12:59:36,624:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "GET /test/index.html HTTP/1.1" 404 -
2020-09-25 12:59:36,633:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "HEAD /test HTTP/1.1" 200 -
2020-09-25 12:59:36,644:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "GET /test?website HTTP/1.1" 200 -
2020-09-25 12:59:36,656:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "GET /test/index.html HTTP/1.1" 404 -
2020-09-25 12:59:36,664:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "HEAD /test HTTP/1.1" 200 -
2020-09-25 12:59:36,675:API: 172.17.0.2 - - [25/Sep/2020 12:59:36] "GET /test?website HTTP/1.1" 200 -
....
```
Where the last three lines just loop endlessly

# Steps to reproduce
- Start localstack
- Create bucket
- Put bucket website
- Get bucket lifecycle (doesn't return)
## Command used to start LocalStack
`DEBUG=1 SERVICES=s3 localstack start`
## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
- `aws --endpoint-url=http://localhost:4566 s3api create-bucket --bucket test`
- `aws --endpoint-url=http://localhost:4566 s3api put-bucket-website --bucket test --website-configuration '{"ErrorDocument": {"Key": "index.html"}, "IndexDocument": {"Suffix": "index.html"}}'`
- `aws --endpoint-url=http://localhost:4566 s3api get-bucket-lifecycle --bucket test `

**Comment by v3rm0n** [comment]
I did some digging and looks like [this](https://github.com/localstack/localstack/blob/master/localstack/services/s3/s3_listener.py#L1209) is where it enters the endless loop.
Should [this condition](https://github.com/localstack/localstack/blob/master/localstack/services/s3/s3_listener.py#L1199) instead of checking for `parsed.query != 'website'` check for all "special" requests (lifecycle, replication, cors etc.) :thinking:

**Comment by whummer** [comment]
Thanks for reporting and providing the fix (in #3030). 👍
